service/fileService.go

```diff
@@ -33,12 +33,13 @@ func Download(link string, episodeTitle string, podcastName string, prefix strin
 		return "", err
 	}
 
-	fileName := getFileName(link, episodeTitle, ".mp3")
-	if prefix != "" {
-		fileName = fmt.Sprintf("%s-%s", prefix, fileName)
-	}
-	folder := createDataFolderIfNotExists(podcastName)
-	finalPath := path.Join(folder, fileName)
+	fileExtension := path.Ext(getFileName(link, episodeTitle, ".mp3"))
+	finalPath := path.Join(
+		os.Getenv("DATA"),
+		cleanFileName(podcastName),
+		fmt.Sprintf("%s%s", episodePathName, fileExtension))
+	dir, _ := path.Split(finalPath)
+	createPreSanitizedPath(dir)
```

Comment:
Mostly use the episodePathName as it is passed in to Download() and don't sanitize the path components again as they were already sanitized as they were created.